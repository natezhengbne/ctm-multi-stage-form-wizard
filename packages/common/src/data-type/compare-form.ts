export type CompareForm = {
  personal?: Personal;
  address?: Address;
};

export type Personal = {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
};

export type Address = {
  streetNumber: string;
  streetName: string;
  streetType: typeof streetTypes[number];
  suburb: string;
  postcode: string;
};

export const streetTypes = [
  "CI", "Ct", "St", "PI", "Ave",
]
