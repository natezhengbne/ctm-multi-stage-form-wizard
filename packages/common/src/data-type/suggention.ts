export type SuburbSuggestions = {
  _embedded: {
    suggestions: Suburb[];
  };
};

export type Suburb = {
  display: {
    text: string;
  };
  source: {
    postcode?: string;
    name?: string;
    state?: string;
  };
};
