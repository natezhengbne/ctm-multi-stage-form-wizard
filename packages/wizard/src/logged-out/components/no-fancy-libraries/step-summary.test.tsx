import { CompareForm } from "@ctm-multi-stage-wizard/common/data-type/compare-form";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import StepPersonal from "./step-personal";
import StepSummary from "./step-summary";

describe("The page of step-summary renders", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<StepSummary step={1} activeStep={1} onPrev={jest.fn()} value={undefined} />);
    expect(wrapper).toMatchSnapshot();
  });
});

// describe("The page of step-summary integration test", () => {
//     let wrapperStepPersonal: ShallowWrapper = shallow(<StepPersonal step={1} activeStep={1} onNext={jest.fn()} onPrev={prevCb} value={undefined} />);
//   let nextCb: jest.Mock;
//   let prevCb: jest.Mock;
//   beforeEach(() => {
//     nextCb = jest.fn((value: CompareForm) => {});
//     prevCb = jest.fn();
//     wrapper 
//   });
// });
