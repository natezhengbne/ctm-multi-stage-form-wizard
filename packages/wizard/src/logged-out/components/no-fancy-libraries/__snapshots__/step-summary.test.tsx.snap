// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page of step-summary renders renders correctly 1`] = `
<Fragment>
  <div
    className="StepperForm-root container-sm"
    id="summary-form"
  >
    <h1>
      Summary
    </h1>
    <div
      className="row g-3"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="firstName"
        >
          First Name
        </label>
        <input
          className="form-control"
          id="firstName"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="lastName"
        >
          Last Name
        </label>
        <input
          className="form-control"
          id="lastName"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          className="form-control"
          id="phone"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-4 mb-3"
      >
        <label
          className="form-label"
          htmlFor="streetNumber"
        >
          Street Number
        </label>
        <input
          className="form-control"
          id="streetNumber"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="streetName"
        >
          Street Name
        </label>
        <input
          className="form-control"
          id="streetName"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-2 mb-3"
      >
        <label
          className="form-label"
          htmlFor="streetType"
        >
          Street Type
        </label>
        <input
          className="form-control"
          id="streetType"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="suburb"
        >
          Suburb
        </label>
        <input
          className="form-control"
          id="suburb"
          readOnly={true}
        />
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <label
          className="form-label"
          htmlFor="postcode"
        >
          Postcode
        </label>
        <input
          className="form-control"
          id="postcode"
          readOnly={true}
        />
      </div>
    </div>
    <div
      className="StepperActions-root"
    >
      <button
        className="btn btn-secondary m-3"
        onClick={[Function]}
      >
        Back
      </button>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</Fragment>
`;
